test/Microsoft.Extensions.Primitives.Tests/StringValuesTests.cs

```diff
@@ -302,5 +302,55 @@ public void Concat(StringValues stringValues, string[] array)
             string[] expectedAppended = filled.Concat(array).ToArray();
             Assert.Equal(expectedAppended, StringValues.Concat(new StringValues(filled), stringValues));
         }
+
+
+        [Fact]
+        public void StringValuesEquals_Works()
+        {
+            // Null strings
+
+            string[] nullStringArray = null;
+            StringValues stringValues = nullStringArray;
+            Assert.Equal(0, stringValues.Count);
+
+            Assert.True(StringValues.Equals(stringValues, (string)null));
+            Assert.True(StringValues.Equals(stringValues, new string[0]));
+            Assert.True(StringValues.Equals(stringValues, (string)stringValues));
+            Assert.True(StringValues.Equals(stringValues, (string[])stringValues));
+            
+            // Empty strings
+
+            string aString = "";
+            stringValues = aString;
+            string[] emptyStringArray = new string[] { aString };
+
+            Assert.True(StringValues.Equals(stringValues, string.Empty));
+            Assert.True(StringValues.Equals(stringValues, aString));
+            Assert.True(StringValues.Equals(stringValues, emptyStringArray));
+
+            // Single strings
+
+            aString = "abc";
+            string[] aStringArray = new[] { aString };
+            stringValues = aStringArray;
+            Assert.Equal(1, stringValues.Count);
+
+            Assert.True(StringValues.Equals(aString, stringValues));
+            Assert.True(StringValues.Equals(aStringArray, stringValues));
+            Assert.True(StringValues.Equals(new StringValues(aString), stringValues));
+            Assert.True(StringValues.Equals(new StringValues(aStringArray), stringValues));
+
+            // String arrays
+
+            aString = "abc";
+            string bString = "bcd";
+            aStringArray = new[] { aString, bString };
+            stringValues = aStringArray;
+            Assert.Equal(2, stringValues.Count);
+
+            Assert.True(StringValues.Equals(aStringArray, stringValues));
+            Assert.True(StringValues.Equals(new StringValues(aStringArray), stringValues));
+            Assert.False(StringValues.Equals(new StringValues(new[] { bString, aString }), stringValues));
```

Comment:
This whole thing should be a data driven test instead of one big blob. Use `[Theory]`